tests/integration/ha_tests/helpers.py

```diff
@@ -759,3 +773,229 @@ async def stop_continuous_writes(ops_test: OpsTest) -> int:
     )
     action = await action.wait()
     return int(action.results["writes"])
+
+
+async def get_storage_id(ops_test: OpsTest, unit_name: str) -> str:
+    """Retrieves  storage id associated with provided unit.
+
+    Note: this function exists as a temporary solution until this issue is ported to libjuju 2:
+    https://github.com/juju/python-libjuju/issues/694
+    """
+    model_name = ops_test.model.info.name
+    proc = subprocess.check_output(f"juju storage --model={model_name}".split())
+    proc = proc.decode("utf-8")
+    for line in proc.splitlines():
+        if "Storage" in line:
+            continue
+
+        if len(line) == 0:
+            continue
+
+        if "detached" in line:
+            continue
+
+        if line.split()[0] == unit_name:
+            return line.split()[1]
+
+
+def is_pods_exists(ops_test: OpsTest, unit_name: str) -> bool:
+    client = Client(namespace=ops_test.model.name)
+    pods = client.list(Pod, namespace=ops_test.model.name)
+
+    for pod in pods:
+        print(
+            f"Pod: {pod.metadata.name} STATUS: {pod.status.phase} TAGGED: {unit_name.replace('/', '-')}"
+        )
+        if (pod.metadata.name == unit_name.replace("/", "-")) and (pod.status.phase == "Running"):
+            return True
+
+    return False
+
+
+async def is_storage_exists(
+    ops_test: OpsTest, storage_id: str, include_detached: bool = False
```

Comment:
The `include_detached` parameter is not being used in this function.

Reply:
fixed